draft-ietf-jsonpath-base.md

```diff
@@ -1361,9 +1362,47 @@ Provisional registration? (standards tree only):
 
 # Security Considerations {#Security}
 
-This section gives security considerations, as required by {{RFC3552}}.
-
-
+Security considerations for JSONPath can stem from
+
+* attack vectors on JSONPath implementations, and
+* the way JSONPath is used in security-relevant mechanisms.
+
+## Attack vectors on JSONPath Implementations
+
+Historically, JSONPath has often been implemented by feeding parts of
+the query to an underlying programming language engine, e.g.,
+JavaScript.
+This approach is well known to lead to injection attacks and would
+require perfect input validation to prevent these attacks (see
+{{Section 12 of -json}} for similar considerations for JSON itself).
+Instead, JSONPath implementations need to implement the entire syntax
+of the query without relying on the parsers of programming language
+engines.
+
+Attacks on availability may attempt to trigger unusually expensive
+runtime performance exhibited by certain implementations in certain
+cases.
+(See {{Section 10 of -cbor}} for issues in hash-table implementations,
+and {{Section 8 of -iregexp}} for performance issues in regular
+expression implementations.)
+Implementers need to be aware that good average performance is not
+sufficient as long as an attacker can choose to submit specially
+crafted JSONPath queries that trigger surprisingly high, possibly
+exponential, CPU usage.
+
+## Attacks on Security Mechanisms that Employ JSONPath
+
+Where JSONPath is used as a part of a security mechanism, attackers
+can attempt to evoke unexpected behavior, or take advantage of
+differences in behavior between implementations that implement some
+protection and the implementations that are being protected by these.
```

Comment:
I don't understand the contrast which is being made here: "implementations that implement some
protection" vs "implementations that are being protected by these [protections?]". Was the following the intended sense?
```suggestion
protection and the implementations that do not.
```

Reply:
The intention was to distinguish the protector from the protected.

Reply:
... if the protector is trying to protect the protected from the presence of certain characteristics, but the protected can be made to see different ones than the protector sees, the protection isn't going to work well.  Tons of CVEs of this class, often called check-use (but different from TOCTOU which is a race condition).

Reply:
Hmmm. It's very confusing. Is the "difference in behaviour" due to the protector and the protected using different implementations of JSONPath?